compiler/rustc_typeck/src/check/method/prelude2021.rs

```diff
@@ -206,6 +210,32 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             return;
         }
 
+        // For from_iter, check if the type actualy implements FromIterator.
+        // If we know it does not, we don't need to warn.
+        if method_name.name == sym::from_iter {
+            if let Some(trait_def_id) = self.tcx.lang_items().from_iterator_trait() {
```

Comment:
This doesn't need to be a lang item, you can use `rustc_diagnostic_item` instead.